From 312429025a3fd94792d0b260493c6835f43e1dfb Mon Sep 17 00:00:00 2001
From: Naga Sureshkumar Relli <naga.sureshkumar.relli@xilinx.com>
Date: Fri, 1 Apr 2016 15:17:38 +0530
Subject: [PATCH 703/793] edac: synopsys: Fix incorrect defines and masks

This patch fixes incorrect DDR control width defines and
also ECC_CEADDR1_BLKNR_MASK value.

Signed-off-by: Naga Sureshkumar Relli <nagasure@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/edac/synopsys_edac.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/edac/synopsys_edac.c b/drivers/edac/synopsys_edac.c
index 97eb29e..5799744 100644
--- a/drivers/edac/synopsys_edac.c
+++ b/drivers/edac/synopsys_edac.c
@@ -149,9 +149,9 @@
 #define ECC_CTRL_CLR_UE_ERRCNT	BIT(3)
 
 /* DDR Control Register width definitions  */
-#define DDRCTL_EWDTH_16		0
+#define DDRCTL_EWDTH_16		2
 #define DDRCTL_EWDTH_32		1
-#define DDRCTL_EWDTH_64		2
+#define DDRCTL_EWDTH_64		0
 
 /* ECC status regsiter definitions */
 #define ECC_STAT_UECNT_MASK	0xF0000
@@ -171,7 +171,7 @@
 #define ECC_CEADDR0_RNK_MASK	BIT(24)
 #define ECC_CEADDR1_BNKGRP_MASK	0x3000000
 #define ECC_CEADDR1_BNKNR_MASK	0x70000
-#define ECC_CEADDR1_BLKNR_MASK	0xF00
+#define ECC_CEADDR1_BLKNR_MASK	0xFFF
 #define ECC_CEADDR1_BNKGRP_SHIFT	24
 #define ECC_CEADDR1_BNKNR_SHIFT	16
 
-- 
1.7.5.4

